back/engines/commercial/insights/app/controllers/insights/web_api/v1/inputs_controller.rb

```diff
@@ -15,6 +15,19 @@ def index
         render json: linked_json(inputs, InputSerializer, serialize_options)
       end
 
+      def index_xlsx
+        # index_xlsx is not policy scoped, instead the view is authorized.
+        inputs = Insights::InputsFinder.new(view, index_xlsx_params).execute
+        categories = view.categories
+        xlsx = xlsx_service.generate_inputs_xlsx inputs,
+                                                 categories,
+                                                 view_private_attributes: Pundit.policy!(current_user,
+                                                                                         User).view_private_attributes?
+
+        send_data xlsx, type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+                        filename: 'inputs.xlsx'
```

Comment:
(nice to have: the filename contains the name of the view (dasherized or snakecase))

Reply:
A better filename is usually computed in the front-end, so skipping here (FE has more info on the context, this could be used in different scenarios in the future)